src/k8s/pkg/k8sd/app/cluster_util.go

```diff
@@ -58,3 +60,22 @@ func waitApiServerReady(ctx context.Context, snap snap.Snap) error {
 
 	return nil
 }
+
+func DetermineLocalhostAddress(clusterMembers []mctypes.ClusterMember) (string, error) {
+	// Check if any of the cluster members have an IPv6 address, if so return "::1"
+	// if one member has an IPv6 address, other members should also have IPv6 interfaces
+	for _, clusterMember := range clusterMembers {
+		memberAddress := clusterMember.Address.Addr().String()
+		nodeIP := net.ParseIP(memberAddress)
+		if nodeIP == nil {
+			return "", fmt.Errorf("failed to parse node IP address %q", memberAddress)
+		}
+
+		if nodeIP.To4() == nil {
+			return "::1", nil
```

Comment:
Shouldn't this be `[::1]`?

Reply:
Cilium seems to add the square brackets on it's own, so having the brackets here is turned into `[[::1]]` which breaks. Updated to return `[::1]` here and trim these on cilium apply instead.